tools/setup.py

```diff
@@ -103,14 +87,96 @@ def check_mongo(self):
+            ZSHRC_PATH = f"{HOME_PATH}/.zshrc"
+            if os.path.isfile(BASHRC_PATH):
+                self.write_alias_to_file(BASHRC_PATH)
+                self.write_alias_to_file('dookie')
+            if os.path.isfile(ZSHRC_PATH):
+                self.write_alias_to_file(ZSHRC_PATH)
+        elif self.operating == "Linux":
+            BASH_PROFILE_PATH = f"{HOME_PATH}/.bash_profile"
+            if os.path.isfile(BASHRC_PATH):
+                self.write_alias_to_file(BASHRC_PATH)
+            elif os.path.isfile(BASH_PROFILE_PATH):
+                self.write_alias_to_file(BASH_PROFILE_PATH)
+
+    def add_alias_windows(self):
+        subprocess.check_call("pip install cx_Freeze",
+                              stderr=subprocess.STDOUT, shell=True)
+        subprocess.check_call("pip install idna",
+                              stderr=subprocess.STDOUT, shell=True)
+        subprocess.check_call("py setup.py build",
+                              stderr=subprocess.STDOUT, shell=True)
+        where_at = os.path.join(os.getcwd(), os.listdir("build")[0])
+        self.color.print_pink(f"""
+HOLD UP!!!
+To use the sce command line tool, add this to your path:
+{where_at}
+                              """, True)
+
+    def add_sce_alias(self):
+        self.check_os()
+        if self.operating == "Windows":
+            self.add_alias_windows()
+        else:
+            self.add_alias_unix()
+
     def setup_core_v4(self):
         """
         This method checks for the corev4 directory
         """
         self.check_directory("Core-v4")
+        os.chdir("Core-v4")
+        subprocess.check_call(
+            "npm install", stderr=subprocess.STDOUT, shell=True)
+        if self.operating == "Windows":
+            subprocess.check_call("py setup.py", stdout=subprocess.PIPE,
+                                  stderr=subprocess.STDOUT, shell=True)
+        else:
+            subprocess.check_call("python3 setup.py", stdout=subprocess.PIPE,
+                                  stderr=subprocess.STDOUT, shell=True)
+        subprocess.check_call("npm run server-install",
+                              stderr=subprocess.STDOUT, shell=True)
+        os.chdir("..")
 
     def setup_discord_bot(self):
         """
         This method checks for the discord bot directory
         """
         self.check_directory("SCE-discord-bot")
+        os.chdir("SCE-discord-bot")
+        subprocess.check_call("npm install",
+                              stderr=subprocess.STDOUT, shell=True)
+        os.chdir("..")
+
+    def setup(self):
+        self.check_os()
+        self.check_mongo()
+
+        self.setup_rpc()
+        self.setup_core_v4()
+        self.setup_discord_bot()
+
+        self.add_sce_alias()
+        self.color.print_yellow("""
+The npm install step in the three projects may have created some unwanted files.
+Open the projects and delete any unfamiliar untracked files.
+                                """)
```

Comment:
Probably include something after this like "Press enter to exit the setup" so the users know that they're done